HubNotificationHandler/src/test/java/com/eagle/hubnotifier/NotifyHandlerServiceImplTest.java

```diff
@@ -50,47 +41,83 @@
+        Map<String, List<String>> recipients = new HashMap<>();
+        Map<String, Object> values = new HashMap<>();
+        NotifyHookServiceImpl notifyHookService = mock(NotifyHookServiceImpl.class);
         NotificationEvent notificationEvent = mock(NotificationEvent.class);
-        doNothing().when(notifyHandlerServiceImpl).sendNotification(notificationEvent);
+        when(notificationEvent.getEventId()).thenReturn(EVENT_ID);
+        when(notificationEvent.getRecipients()).thenReturn(recipients);
+        when(notificationEvent.getRootOrg()).thenReturn(ROOT_ORG);
+        when(notificationEvent.getTagValues()).thenReturn(values);
+        when(notificationEvent.getTargetData()).thenReturn(values);
+        when(configuration.getDiscussionCreateUrl()).thenReturn(DISCUSSION_BASE_URL);
+        when(hubTopicRepository.findByKey(TOPIC_DATA)).thenReturn(topic);
+        when(userRepository.findByKey(USER_CONST)).thenReturn(user);
+        doNothing().when(notifyHookService).handleTopicReplyEvent(Matchers.any(Map.class));
+    }
+
+    @Test
+    void handleTopicUpvoteEvent() {
+        HubPost post = mock(HubPost.class);
+        HubUser user = mock(HubUser.class);
+        NotifyHookServiceImpl notifyHookService = mock(NotifyHookServiceImpl.class);
+        when(hubPostRepository.findByKey(POST_DATA)).thenReturn(post);
+        List<HubUser> userList = new ArrayList<>();
+        userList.add(user);
+        when(userRepository.findByUUIDS(Arrays.asList("1", "2"))).thenReturn(userList);
+        doNothing().when(notifyHookService).handleTopicUpvoteEvent(Matchers.any(Map.class));
     }
 
     @Test
-    void OutBoundGetRequest() {
+    void handleTopicDownVoteEvent() {
+        HubPost post = mock(HubPost.class);
+        HubUser user = mock(HubUser.class);
+        NotifyHookServiceImpl notifyHookService = mock(NotifyHookServiceImpl.class);
+        when(hubPostRepository.findByKey(POST_DATA)).thenReturn(post);
+        List<HubUser> userList = new ArrayList<>();
+        userList.add(user);
+        when(userRepository.findByUUIDS(Arrays.asList("1", "2"))).thenReturn(userList);
+        doNothing().when(notifyHookService).handleTopicDownVoteEvent(Matchers.any(Map.class));
+    }
+
+    @Test
+    void sendNotification() {
         NotificationEvent notificationEvent = mock(NotificationEvent.class);
-        StringBuilder builder = new StringBuilder();
-        builder.append("http//localhost:9090/notifyHandler");
-        ResponseEntity entity  =  (ResponseEntity) outboundRequestHandlerService.fetchResultUsingPost(builder, notificationEvent);
-        assertTrue(entity!=null);
-        assertTrue(entity.getStatusCode().equals(HttpStatus.INTERNAL_SERVER_ERROR));
+        when(configuration.getNotifyServiceHost()).thenReturn(NOTIFICATION_SERVICE_HOST);
+        when(configuration.getNotifyServicePath()).thenReturn(NOTIFICATION_SERVICE_PATH);
+        doNothing().when(notifyHandlerServiceImpl).sendNotification(notificationEvent);
```

Comment:
Please look into this again. using this code it will not do any operation when notifyHandlerServiceImpl class will be called.